clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/GCEUtil.groovy

```diff
@@ -188,6 +190,30 @@ class GCEUtil {
     }
   }
 
+  static def queryHealthCheck(String projectName, String healthCheckName, Compute compute, Task task, String phase) {
+    task.updateStatus phase, "Looking up http(s) health check $healthCheckName..."
+
+    try {
+      return compute.httpHealthChecks().get(projectName, healthCheckName).execute()
+    } catch (GoogleJsonResponseException e) {
+      // 404 is thrown, and the details are populated, if the health check does not exist in the given project.
+      // Any other exception should be propagated directly.
+      if (e.getStatusCode() == 404 && e.details) {
+        try {
+          return compute.httpsHealthChecks().get(projectName, healthCheckName).execute()
```

Comment:
If you're checking for flakiness here, should we use some sort of backoff/sleep before making a second identical request?